parser-typechecker/src/Unison/Codebase/Editor/Propagate.hs

```diff
@@ -204,14 +233,20 @@ propagate patch b = case validatePatch patch of
             seen' = seen <> Set.fromList (view _1 . view _2 <$> joinedStuff)
             writeTypes =
               traverse_ (\(Reference.DerivedId id, tp) -> eval $ PutDecl id tp)
-          writeTypes (Map.elems componentMap)
+            constructorMapping =
+              constructorReplacements
+                <> generateConstructorMapping componentMap hashedComponents'
+          traceM $ "Better not be empty: " <> show constructorMapping
+          traceM $ "Better not be empty either: " <> show hashedComponents'
+          writeTypes $ Map.toList newNewTypes
```

Comment:
This value `newNewTypes` now has all the brand new types we want to actually write to the codebase at this point.